src/main/java/bio/terra/workspace/service/resource/controlled/flight/delete/DeleteControlledResourceGcsBucketFlight.java

```diff
@@ -36,55 +42,115 @@ public DeleteControlledResourceGcsBucketFlight(FlightMap inputParameters, Object
     super(inputParameters, beanBag);
     final FlightBeanBag flightBeanBag = FlightBeanBag.getFromObject(beanBag);
 
+    final UUID workspaceId = inputParameters.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    final UUID resourceId =
+        inputParameters.get(WorkspaceFlightMapKeys.ResourceKeys.RESOURCE_ID, UUID.class);
+    final AuthenticatedUserRequest userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+
     // Flight plan:
-    // 1. TODO: Delete the Sam resource - this should make the object invisible via get and
-    // enumerate endpoints
+    // 1. Delete the Sam resource. That will make the object inaccessible.
     // 2. Delete the cloud resource:
     //    a. Set the lifecycle on the bucket to delete immediately
     //    b. Try deleting the bucket
     //    c. If delete succeeds, finish step
     //    d. If delete fails, sleep one hour; goto (either a or b; maybe a for belts and suspenders)
     // 3. Delete the metadata
+    addStep(
+        new DeleteSamResourceStep(
+            flightBeanBag.getResourceDao(),
+            flightBeanBag.getSamService(),
+            workspaceId,
+            resourceId,
+            userRequest));
     addStep(
         new DeleteGcsBucketStep(
             flightBeanBag.getCrlService(),
             flightBeanBag.getResourceDao(),
-            flightBeanBag.getWorkspaceService()));
-    addStep(new DeleteMetadataStep(flightBeanBag.getResourceDao()));
+            flightBeanBag.getWorkspaceService(),
+            workspaceId,
+            resourceId));
+    addStep(new DeleteMetadataStep(flightBeanBag.getResourceDao(), workspaceId, resourceId));
+  }
+
+  static class DeleteSamResourceStep implements Step {
+    private final ResourceDao resourceDao;
+    private final SamService samService;
+    private final UUID workspaceId;
+    private final UUID resourceId;
+    private final AuthenticatedUserRequest userRequest;
+
+    // TODO: this looks generic, so factor it out and share it
+    public DeleteSamResourceStep(
+        ResourceDao resourceDao,
+        SamService samService,
+        UUID workspaceId,
+        UUID resourceId,
+        AuthenticatedUserRequest userRequest) {
+      this.resourceDao = resourceDao;
+      this.samService = samService;
+      this.workspaceId = workspaceId;
+      this.resourceId = resourceId;
+      this.userRequest = userRequest;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext flightContext)
+        throws InterruptedException, RetryException {
+
+      WsmResource wsmResource = resourceDao.getResource(workspaceId, resourceId);
+      ControlledResource resource = wsmResource.castToControlledResource();
+
+      try {
+        samService.deleteControlledResource(resource, userRequest);
+      } catch (SamApiException samApiException) {
+        if (samApiException.getApiExceptionStatus() == HttpStatus.NOT_FOUND.value()) {
```

Comment:
nit: might be worth a comment explaining this is for step idempotency

Reply:
Good point. I wrote up the two cases that would cause not found.